lib/gossip/damper.js

```diff
@@ -0,0 +1,580 @@
+                local: self.ringpop.whoami(),
+                dampReqMembers: dampReqMemberAddrs,
+                results: res
+            });
+            self.emit('event', new events.DampingInconclusiveEvent());
+            callback();
+            return;
+        }
+
+        for (var i = 0; i < membersToDamp.length; i++) {
+            self.dampMember(membersToDamp[i]);
+        }
+
+        self.ringpop.stat('increment', 'damper.damp-req.damped');
+        self.logger.warn('ringpop damped members', {
+            local: self.ringpop.whoami(),
+            dampReqMembers: dampReqMemberAddrs,
+            membersToDamp: membersToDamp,
+            results: res
+        });
+        self.emit('event', new events.DampedEvent());
+        callback();
+    }
+};
+
+Damper.prototype.removeFlapper = function removeFlapper(flapper) {
+    var address = flapper.address || flapper;
+    if (!this.flappers[address]) {
+        this.logger.debug('ringpop damper did not remove flapper; it was not added. it may be damped.', {
+            local: this.ringpop.whoami(),
+            flapper: address,
+            isDamped: !!this.dampedMembers[address]
+        });
+        return false;
+    }
+
+    delete this.flappers[address];
+    this.logger.debug('ringpop damper removed flapper', {
+        local: this.ringpop.whoami(),
+        flapper: address
+    });
+    this.ringpop.stat('increment', 'damper.flapper.removed');
+    this.ringpop.stat('gauge', 'damper.flappers',
+        Object.keys(this.flappers).length);
+
+    if (this._getFlapperCount() === 0) {
+        this.logger.debug('ringpop damper stopping damp timer; no flappers left', {
+            local: this.ringpop.whoami()
+        });
+        this._stopDampTimer();
+    }
+
+    return true;
+};
+
+Damper.prototype._fanoutDampReqs = function _fanoutDampReqs(flapperAddrs, dampReqMembers, rVal, callback) {
+    var self = this;
+
+    // Send out the damp-req to each member selected.
+    var request = new DampReqRequest(this.ringpop, flapperAddrs);
+    for (var i = 0; i < dampReqMembers.length; i++) {
+        var dampReqAddr = dampReqMembers[i].address;
+        this.ringpop.stat('increment', 'damp-req.send');
+        this.ringpop.client.protocolDampReq({
+            host: dampReqAddr,
+            timeout: this.ringpop.config.get('dampReqTimeout')
+        }, request,
+            createDampReqHandler(dampReqAddr));
+    }
+
+    var numPendingReqs = dampReqMembers.length;
+    var errors = [];
+    var results = [];
+
+    // Accumulate responses until rVal is satisfied or is impossible to satisfy because
+    // too many error responses.
+    function createDampReqHandler(addr) {
+        var start = Date.now();
+        return function onDampReq(err, res) {
+            var timing = Date.now() - start;
+            self.ringpop.stat('timing', 'protocol.damp-req', timing);
+
+            // Prevents double-callback
+            if (typeof callback !== 'function') return;
```

Comment:
afaik, this violates or normal style guides. regardless of that, I would argue that return always deserves its own space, so that it's very clear that it's an exit from the current scope.